spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/RewriteViewCommands.scala

```diff
@@ -149,4 +163,20 @@ case class RewriteViewCommands(spark: SparkSession) extends Rule[LogicalPlan] wi
         None
     }
   }
+
+  /**
+   * Collect the names of all temporary functions.
+   */
+  private def collectTemporaryFunctions(child: LogicalPlan): Seq[String] = {
+    val tempFunctions = new mutable.HashSet[String]()
+    child.resolveExpressionsWithPruning(_.containsAnyPattern(UNRESOLVED_FUNCTION)) {
+      case f @ UnresolvedFunction(nameParts, _, _, _, _) if isTempFunction(nameParts) =>
+        tempFunctions += nameParts.asFunctionIdentifier.funcName
```

Comment:
I don't think it is a good idea to convert to a function identifier. We can just convert the name parts directly to a string. Plus, we know that if `isTempFunction` is true then it is only one part anyway, so the extra conversion doesn't help make this safe.